Source/WebCore/dom/Element.cpp

```diff
@@ -4434,9 +4434,19 @@ void Element::webkitRequestFullscreen()
     requestFullscreen({ }, nullptr);
 }
 
-// FIXME: Options are currently ignored.
-void Element::requestFullscreen(FullscreenOptions&&, RefPtr<DeferredPromise>&& promise)
-{
+// FIXME: Only KeyboardLock option is currently considered.
+void Element::requestFullscreen(FullscreenOptions&& options, RefPtr<DeferredPromise>&& promise)
+{
+    bool optionsEnabled = document().settings().fullScreenKeyboardLock();
+    if (optionsEnabled)
+        protectedDocument()->fullscreenManager().setKeyboardLockMode(options.keyboardLock);
+    else {
+        if (options.keyboardLock != FullscreenOptions::KeyboardLock::None) {
+            promise->reject(ExceptionCode::NotSupportedError, "options.keyboardLock is unavailable."_s);
+            return;
+        }
+    }
```

Comment:
```suggestion
void Element::requestFullscreen(FullscreenOptions&& options, RefPtr<DeferredPromise>&& promise)
{
    bool optionsEnabled = document().settings().fullScreenKeyboardLock();
    if (optionsEnabled)
        protectedDocument()->fullscreenManager().setKeyboardLockMode(options.keyboardLock);
    else
        ASSERT(options.keyboardLock == FullscreenOptions::KeyboardLock::None);
```

Reply:
@nt1m I do not think the ASSERT can generate promise rejection event on the non KeyboardLock mode.
So I would like to keep the current implementation AS-IS unless there is a way to auto generate the event through the ASSERT statement. How do you think?